pkg/updater/oci.go

```diff
@@ -0,0 +1,138 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package updater
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path"
+
+	"github.com/opencontainers/go-digest"
+	ociSpec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+const (
+	// mediaTypeImageLayerXz is the media type for an OCI image layer in xz format
+	// It's not officially declared by the spec so we do it here
+	mediaTypeImageLayerXz = "application/vnd.oci.image.layer.v1.tar+xz"
+)
+
+// extractOCI extracts the OCI archive at `ociArchivePath`
+func extractOCI(ociArchivePath string, destinationPath string) error {
+	index := &ociSpec.Index{}
+	indexFile, err := os.Open(path.Join(ociArchivePath, "index.json"))
+	if err != nil {
+		return fmt.Errorf("could not open index file: %w", err)
+	}
+	err = json.NewDecoder(indexFile).Decode(index)
+	if err != nil {
+		return fmt.Errorf("could not parse index file: %w", err)
+	}
+
+	for _, manifest := range index.Manifests {
+		if err := extractOCIManifest(ociArchivePath, destinationPath, manifest); err != nil {
```

Comment:
Nit: You're mixing `if err := ... ; err != nil` and `if err != nil` everywhere here 😄

Reply:
Personally I don't like the combined form, it's the more uncommon one and harder to read

Reply:
[5a7b102](https://github.com/DataDog/datadog-agent/pull/22812/commits/5a7b10258f00ce8584654c82aaf1895633b865f7)